### DISCUSSION TITLE
How do I throw RecognitionException when some token is encountered

### DISCUSSION THREAD

**Discussion by nprime496** [antlr/antlr4#3496] [Q&A]
Hi, I've added some rules in my lexer .g4 file but I'm unable to find a good way to throw an exception (to stop parsing) when some token is recognized using ANTR4.

```java
NUM : (DIGIT)+ { /* throw an exception*/};
```
Please how can I do that ?

**Comment by jgremmen** [comment]
Please look at the discussion #3162 I started a while ago. 
Short answer to your question: there is no good way yet.
